storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/utils/LogviewerResponseBuilder.java

```diff
@@ -76,24 +76,18 @@ public static Response buildSuccessJsonResponse(Object entity, String callback,
      * Build a Response object representing download a file.
      *
      * @param host host address
+     * @param contentDispositionName The name to set in the Content-Disposition header
      * @param file file to download
      */
-    public static Response buildDownloadFile(String host, File file, Meter numFileDownloadExceptions) throws IOException {
-        String fname;
-        try {
-            String topoInfo = file.getParentFile().getParentFile().getName();
-            String port = file.getParentFile().getName();
-            fname = String.format("%s-%s-%s-%s", host, port, topoInfo, file.getName());
-        } catch (NullPointerException e) {
-            fname = file.getName();
-        }
+    public static Response buildDownloadFile(String host, String contentDispositionName,
```

Comment:
The `String host` parameter can be removed.

Reply:
Good catch, fixed